case_studies/galaxy_clustering/variational_dist.py

```diff
@@ -38,46 +32,11 @@ def sample(self, use_mode=False) -> TileCatalog:
         Returns:
             TileCatalog: Sampled catalog
         """
-        q = self.factors
-
-        est_cat = {}
-
-        locs = q["loc"].mode if use_mode else q["loc"].sample().squeeze(0)
-        est_cat = {"locs": locs}
-
-        # populate catalog with per-band (log) star fluxes
-        sf_factors = [q[f"star_flux_{band}"] for band in self.survey_bands]
-        sf_lst = [p.mode if use_mode else p.sample() for p in sf_factors]
-        est_cat["star_fluxes"] = torch.stack(sf_lst, dim=3)
-
-        # populate catalog with source type
-        galaxy_bools = q["galaxy_prob"].mode if use_mode else q["galaxy_prob"].sample()
-        galaxy_bools = galaxy_bools.unsqueeze(3)
-        star_bools = 1 - galaxy_bools
-        est_cat["source_type"] = SourceType.STAR * star_bools + SourceType.GALAXY * galaxy_bools
-
-        # populate catalog with galaxy parameters
-        gs_dists = [q[f"galsim_{name}"] for name in self.GALSIM_NAMES]
-        gs_param_lst = [d.icdf(torch.tensor(0.5)) if use_mode else d.sample() for d in gs_dists]
-        est_cat["galaxy_params"] = torch.stack(gs_param_lst, dim=3)
-
-        # populate catalog with per-band galaxy fluxes
-        gf_dists = [q[f"galaxy_flux_{band}"] for band in self.survey_bands]
-        gf_lst = [d.icdf(torch.tensor(0.5)) if use_mode else d.sample() for d in gf_dists]
-        est_cat["galaxy_fluxes"] = torch.stack(gf_lst, dim=3)
+        est_cat = super().sample(use_mode=use_mode)
 
+        q = self.factors
         est_cat["membership"] = q["mem_prob"].mode if use_mode else q["mem_prob"].sample()
-        est_cat["membership"] = est_cat["membership"].unsqueeze(3)
-
-        # we have to unsqueeze these tensors because a TileCatalog can store multiple
-        # light sources per tile, but we sample only one source per tile
-        for k, v in est_cat.items():
-            est_cat[k] = v.unsqueeze(3)
-
-        # n_sources is not unsqueezed because it is a single integer per tile regardless of
-        # how many light sources are stored per tile
-
-        est_cat["n_sources"] = q["on_prob"].mode if use_mode else q["on_prob"].sample()
+        est_cat["membership"] = est_cat["membership"].unsqueeze(3).unsqueeze(3)
```

Comment:
@kapnadak  I updated the syntax for your case study, though I haven't tested it. I think that by calling `super().sample(...)` rather than duplicating code, your case study will be compatible with the changes in this PR.